static/js/common/upload-addon.js

```diff
@@ -355,7 +358,7 @@
                         }
                       } else {  // This is a listed add-on.
                         if (automaticValidation && results.beta) {
-                          if (warnings > 0) {
+                          if (!passedValidation) {
```

Comment:
Regular warnings that should fail an automatic validation (that is used for unlisted non-sideload addons AND for beta files) have a signing severity level attached. So if a message has a signing severity above `trivial`, it'll fail the auto validation.